cmd/fluxmon/main.go

```diff
@@ -0,0 +1,164 @@
+	bindLocalFlag(rootCmd, nameParam, defaultName, `Name for the prometheus variable`)
+	viper.AutomaticEnv() // read in environment variables that match
+}
+
+func bindLocalFlag(c *cobra.Command, name string, value string, help string) {
+	c.Flags().String(name, value, help)
+	viper.BindPFlag(name, c.Flags().Lookup(name))
+}
+
+var rootCmd = &cobra.Command{
+	Use:   "fluxmon",
+	Short: "Monitor a database and expose metrics for prometheus",
+	Long:  `This service will monitor a database for specified queries and expose them to prometheus`,
+	Run: func(cmd *cobra.Command, args []string) {
+		// Logger component.
+		var logger log.Logger
+		{
+			logger = log.NewLogfmtLogger(os.Stderr)
+			logger = log.NewContext(logger).With("ts", log.DefaultTimestampUTC)
+			logger = log.NewContext(logger).With("caller", log.DefaultCaller)
+		}
+
+		// Parse queries
+		queryBytes, err := ioutil.ReadFile(viper.GetString(queriesParam))
+		if err != nil {
+			logger.Log("stage", "read queries", "err", err)
+			os.Exit(1)
+		}
+		var queries queries
+		err = yaml.Unmarshal(queryBytes, &queries)
+		if err != nil {
+			logger.Log("stage", "read queries", "err", err)
+			os.Exit(1)
+		}
+		logger.Log("stage", "read queries", "queries", fmt.Sprintf("%v", len(queries.Queries)))
+
+		// Initialise database driver
+		var dbDriver string
+		{
+			var version uint64
+			u, err := url.Parse(viper.GetString(databaseSourceParam))
+			if err != nil {
+				logger.Log("stage", "db init", "err", err)
+				os.Exit(1)
+			}
+			logger.Log("stage", "db init", "url", u, "scheme", u.Scheme)
+			dbDriver = db.DriverForScheme(u.Scheme)
+			logger.Log("stage", "db init", "driver", dbDriver, "db-version", fmt.Sprintf("%d", version))
+		}
+
+		// Connect to Job store.
+		conn, err := sql.Open(dbDriver, viper.GetString(databaseSourceParam))
+		if err != nil {
+			logger.Log("stage", "db init", "err", err)
+			os.Exit(1)
+		}
+
+		// Prometheus gauge
+		jobStatus := prometheus.NewGaugeFrom(stdprometheus.GaugeOpts{
+			Namespace: viper.GetString(namespaceParam),
+			Subsystem: viper.GetString(subsystemParam),
+			Name:      viper.GetString(nameParam),
+			Help:      "Gauge for database count",
+		}, []string{LabelName})
+
+		// Error channel
+		errc := make(chan error)
+
+		// Start tick
+		logger.Log("stage", "ticker", "period", viper.GetDuration(tickParam))
+		dbTicker := time.NewTicker(viper.GetDuration(tickParam))
+		defer dbTicker.Stop()
+		go func(tick <-chan time.Time) {
+			for range tick {
+				for _, q := range queries.Queries {
+					var count int
+					err = conn.QueryRow(q.Query).Scan(&count)
+					if err != nil {
+						logger.Log("stage", "query", "name", q.Name, "query", q.Query, "err", err)
+						errc <- err
+					}
+					logger.Log("stage", "query", "name", q.Name, "result", fmt.Sprintf("%v", count))
+					jobStatus.With(
+						LabelName, q.Name,
+					).Set(float64(count))
+				}
```

Comment:
Bit confused here.

I thought we wanted a separate process that would monitor the DB and expose some metrics to prom. Let me review the issue again.